src/inference/Inference.jl

```diff
@@ -439,95 +352,54 @@ function AbstractMCMC.bundle_samples(
     return MCMCChains.Chains(
         parray,
         nms,
-        deepcopy(TURING_INTERNAL_VARS);
+        (internals = extra_params,);
         evidence=le,
         info=info,
     ) |> sort
 end
 
 # This is type piracy (for SampleFromPrior).
 function AbstractMCMC.bundle_samples(
-    rng::AbstractRNG,
+    ts::Vector,
     model::AbstractModel,
     spl::Union{Sampler{<:InferenceAlgorithm},SampleFromPrior},
-    N::Integer,
-    ts::Vector,
+    state,
     chain_type::Type{Vector{NamedTuple}};
-    discard_adapt::Bool=true,
-    save_state=false,
     kwargs...
 )
-    nts = Vector{NamedTuple}(undef, N)
-
-    for (i, t) in enumerate(ts)
-        params = getparams(t)
-
-        k = collect(keys(params))
-        vs = []
-        for v in values(params)
-            push!(vs, v[1])
-        end
-
-        push!(k, :lp)
-
-        nts[i] = NamedTuple{tuple(k...)}(tuple(vs..., getlogp(t)))
+    return map(ts) do t
+        params = map(first, getparams(t))
+        return merge(params, (lp = getlogp(t),))
```

Comment:
Couldn't we use `metadata(t)` here?

Reply:
Sure, we can. I guess I didn't want to change the current behaviour but maybe it is actually better (and more consistent with the Chains output) to add the full metadata?

Reply:
Yeah, it'll prevent future errors I think if we start to lean more heavily on `metadata` (which is correct, IMO).

```suggestion
        return merge(params, metadata(t))
```